### PR TITLE
Add first pass at OAuth2 authentication to endpoints

### PR DESCRIPTION
**Pull request by rjw57**
This PR is best reviewed commit-wise. 2cb9807 and dc8f851 do some plumbing required to implement the meat of the change. 2cb9807 adds a hydra-base OAuth2 token issuer to the docker-compose framework for easier testing during development

The main body of the PR is represented by 352cf80 which adds OAuth2 authentication. See the commit message for full details.

**Update:** 21def11 was added to fix Travis tests.

### CODE REVIEW

**Comment by rjw57** [comment]
Closed for the moment while we figure out an answer to the lookup requirement.

**Comment by rjw57** [comment]
I've added 21def11 which fixes the Travis checks at the expense of removing ucamlookup from ``INSTALLED_APPS``.

File: assets/test_views.py [hunk 1]
```diff
@@ -1,8 +1,13 @@
 import copy
```
Comment (msb) [inline assets/test_views.py]: s/b in tests dir

Comment (rjw57) [reply assets/test_views.py]: S/b?

**Pull request merged**